e2e/explorer/workflow/run.spec.ts

```diff
@@ -0,0 +1,288 @@
+import { createNamespace, deleteNamespace } from "../../utils/namespace";
+import { expect, test } from "@playwright/test";
+
+import { noop as basicWorkflow } from "~/pages/namespace/Explorer/Tree/NewWorkflow/templates";
+import { createWorkflow } from "~/api/tree/mutate/createWorkflow";
+import { faker } from "@faker-js/faker";
+import { getInput } from "~/api/instances/query/input";
+import { jsonSchemaFormWorkflow } from "./utils";
+
+let namespace = "";
+
+test.beforeEach(async () => {
+  namespace = await createNamespace();
+});
+
+test.afterEach(async () => {
+  await deleteNamespace(namespace);
+  namespace = "";
+});
+
+test("it is possible to open and use the run workflow modal from the editor and the header of the workflow page", async ({
+  page,
+}) => {
+  const workflow = faker.system.commonFileName("yaml");
+  await createWorkflow({
+    payload: basicWorkflow.data,
+    headers: undefined,
+    urlParams: {
+      baseUrl: process.env.VITE_DEV_API_DOMAIN,
+      namespace,
+      name: workflow,
+    },
+  });
+
+  await page.goto(`${namespace}/explorer/workflow/active/${workflow}`);
+
+  // open modal via editor button
+  await page.getByTestId("workflow-editor-btn-run").click();
+  expect(
+    await page.getByTestId("run-workflow-dialog"),
+    "it opens the dialog from the editor button"
+  ).toBeVisible();
+  await page.getByTestId("run-workflow-cancel-btn").click();
+  expect(await page.getByTestId("run-workflow-dialog")).not.toBeVisible();
+
+  // open modal via header button
+  await page.getByTestId("workflow-header-btn-run").click();
+  expect(
+    await page.getByTestId("run-workflow-dialog"),
+    "it opens the dialog from the header button"
+  ).toBeVisible();
+
+  // use the tabs
+  expect(
+    await page
+      .getByTestId("run-workflow-json-tab-btn")
+      .getAttribute("aria-selected"),
+    "the json tab is selected by default (since this workflow has no JSON schema)"
+  ).toBe("true");
+
+  expect(
+    await page
+      .getByTestId("run-workflow-form-tab-btn")
+      .getAttribute("aria-selected")
+  ).toBe("false");
+
+  await page.getByTestId("run-workflow-form-tab-btn").click();
+
+  expect(
+    await page
+      .getByTestId("run-workflow-form-tab-btn")
+      .getAttribute("aria-selected"),
+    "the form tab is now selected"
+  ).toBe("true");
+
+  expect(
+    await page
+      .getByTestId("run-workflow-json-tab-btn")
+      .getAttribute("aria-selected")
+  ).toBe("false");
+
+  expect(
+    await page.getByTestId("run-workflow-form-input-hint"),
+    "it shows a hint that no form could be generated"
```

Comment:
This was a typo and [i just changed it](https://github.com/direktiv/direktiv-ui/pull/402/commits/dcbb2cdd4d1351510ab8c1c9431a3efc6668dfbe) to be `toBeVisible()`. Funny enough, both work. I had some similar issues like this before, which seem counterintuitive at first glance. I think it is not visible at first and then becomes visible just a few milliseconds later.

So when the test says wait for it to **be not visible** it resolves immediately, if it says wait for it to **be visible** it waits a few milliseconds longer and then resolves. 

I wonder if it is just bad practice to use `not.toBeVisible()` at all, because you never know if it will become visible immediately after that line.